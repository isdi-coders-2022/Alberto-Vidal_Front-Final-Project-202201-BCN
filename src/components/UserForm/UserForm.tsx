/* eslint-disable react/jsx-props-no-spreading */
import { Button, TextField } from "@mui/material";
import { Controller, useForm } from "react-hook-form";
import styled from "styled-components";

const Form = styled.form`
  display: flex;
  margin: 0 auto;
  width: 400px;
  height: 600px;
  margin-top: 30px;
  flex-direction: column;
  justify-content: space-evenly;
  display: flex;
  & button {
    align-self: flex-start;
    &:active {
      transition: 0.3s all;
      transform: translateY(3px);
      opacity: 0.8;
    }
  }
`;

const UserForm = ({ isLogin }: UserFormProps): JSX.Element => {
  const {
    control,
    handleSubmit,
    formState: { dirtyFields },
  } = useForm<LoginUserFormData | RegisterUserFormData>();

  return (
    <Form onSubmit={handleSubmit((data) => data)}>
      <Controller
        name="username"
        defaultValue=""
        control={control}
        render={({ field }) => (
          <TextField variant="outlined" {...field} label={field.name} />
        )}
      />
      <Controller
        name="password"
        control={control}
        defaultValue=""
        render={({ field }) => (
          <TextField variant="outlined" {...field} label={field.name} />
        )}
      />
      {!isLogin && (
        <>
          <Controller
            name="password"
            control={control}
            defaultValue=""
            render={({ field }) => (
              <TextField variant="outlined" {...field} label={field.name} />
            )}
          />
          <Controller
            name="avatar"
            control={control}
            defaultValue=""
            render={({ field }) => (
              <TextField variant="outlined" {...field} label={field.name} />
            )}
          />
        </>
      )}
      <Button
        variant="contained"
        type="submit"
        disabled={!(dirtyFields.username && dirtyFields.password)}
      >
        Submit
      </Button>
    </Form>
  );
};

export default UserForm;

interface LoginUserFormData {
  username: string;
  password: string;
}

interface RegisterUserFormData {
  username: string;
  name: string;
  password: string;
  avatar: string;
}

interface UserFormProps {
  isLogin: boolean;
}
