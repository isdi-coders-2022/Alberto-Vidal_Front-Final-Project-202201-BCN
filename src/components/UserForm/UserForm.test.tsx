import { render, screen, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { BrowserRouter } from "react-router-dom";
import UserForm from "./UserForm";

describe("Given a UserForm component", () => {
  describe("When it's rendered with prop isLogin set to true", () => {
    test("Then it should render username and password fields", () => {
      const onSubmit = jest.fn();
      const isLogin = true;
      const expectedFields = [/username/i, /password/i];

      render(
        <BrowserRouter>
          <UserForm isLogin={isLogin} onSubmit={onSubmit} />
        </BrowserRouter>
      );
      const fields = expectedFields.map((label) =>
        screen.getByLabelText(label)
      );

      fields.forEach((field) => expect(field).toBeInTheDocument());
    });

    test("Then it should render only two fields", () => {
      const onSubmit = jest.fn();
      const isLogin = true;

      render(
        <BrowserRouter>
          <UserForm isLogin={isLogin} onSubmit={onSubmit} />
        </BrowserRouter>
      );
      const fields = screen.getByRole("textbox");
      const password = screen.getByLabelText("password");

      expect(fields).toBeInTheDocument();
      expect(password).toBeInTheDocument();
    });
  });

  describe("When it's rendered with prop isLogin set to false", () => {
    test("Then it should render username, password, name, and avatar fields", () => {
      const onSubmit = jest.fn();
      const isLogin = false;
      const expectedFields = [/username/i, /password/i, /^name/i, /avatar/i];

      render(
        <BrowserRouter>
          <UserForm isLogin={isLogin} onSubmit={onSubmit} />
        </BrowserRouter>
      );
      const fields = expectedFields.map((label) =>
        screen.getByLabelText(label)
      );

      fields.forEach((field) => expect(field).toBeInTheDocument());
    });
  });
  describe("When it's rendered with prop isLogin set to false and all the inputs filled", () => {
    test("Then the button shouldnt be disabled", () => {
      const onSubmit = jest.fn();
      const isLogin = false;
      const expectedFields = [/username/i, /password/i, /^name/i, /avatar/i];
      const buttonName = isLogin ? /log in/i : /register/i;
      const someText = "tueanhnu";

      render(
        <BrowserRouter>
          <UserForm isLogin={isLogin} onSubmit={onSubmit} />
        </BrowserRouter>
      );
      const fields = expectedFields.map((label) =>
        screen.getByLabelText(label)
      );
      fields.forEach((field) => userEvent.type(field, someText));
      const button = screen.getByRole("button", { name: buttonName });

      expect(button).not.toBeDisabled();
    });
  });

  describe("When it's rendered with prop isLogin set to true and all the inputs filled", () => {
    test("Then the button shouldnt be disabled", () => {
      const onSubmit = jest.fn();
      const isLogin = true;
      const expectedFields = [/username/i, /password/i];
      const buttonName = isLogin ? /log in/i : /register/i;
      const someText = "tueanhnu";

      render(
        <BrowserRouter>
          <UserForm isLogin={isLogin} onSubmit={onSubmit} />
        </BrowserRouter>
      );
      const fields = expectedFields.map((label) =>
        screen.getByLabelText(label)
      );
      fields.forEach((field) => userEvent.type(field, someText));
      const button = screen.getByRole("button", { name: buttonName });

      expect(button).not.toBeDisabled();
    });
  });

  describe("When it's rendered with prop isLogin set to false and all the inputs filled and clicked on the button", () => {
    test("Then the onSubmit action should be performed", async () => {
      const onSubmit = jest.fn();
      const isLogin = false;
      const expectedFields = [/username/i, /password/i, /^name/i, /avatar/i];
      const buttonName = isLogin ? /log in/i : /register/i;
      const someText = "tueanhnu";

      render(
        <BrowserRouter>
          <UserForm isLogin={isLogin} onSubmit={onSubmit} />
        </BrowserRouter>
      );
      const fields = expectedFields.map((label) =>
        screen.getByLabelText(label)
      );
      fields.forEach((field) => userEvent.type(field, someText));
      const button = screen.getByRole("button", { name: buttonName });
      userEvent.click(button);

      await waitFor(() => expect(onSubmit).toHaveBeenCalled());
    });
  });
});
