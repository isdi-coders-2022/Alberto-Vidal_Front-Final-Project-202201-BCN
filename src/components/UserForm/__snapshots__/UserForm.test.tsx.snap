// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a UserForm component When it's rendered with islogin false Then it should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="sc-bdvvtL cRXLZB"
      >
        <label
          class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
          role="button"
          tabindex="0"
        >
          Avatar (Upload File)
          <input
            hidden=""
            id="avatar"
            type="file"
          />
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </label>
        <div
          class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="name"
            id="name-label"
          >
            name
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id="name"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-1g8cwq9"
              >
                <span>
                  name
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="username"
            id="username-label"
          >
            username
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id="username"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-1g8cwq9"
              >
                <span>
                  username
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="password"
            id="password-label"
          >
            password
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id="password"
              type="password"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-1g8cwq9"
              >
                <span>
                  password
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="sc-gsDKAQ hmwpO"
        >
          <a
            href="/login"
          >
            already registred?
          </a>
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root Mui-disabled  css-sghohy-MuiButtonBase-root-MuiButton-root"
            disabled=""
            tabindex="-1"
            type="submit"
          >
            register
          </button>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="sc-bdvvtL cRXLZB"
    >
      <label
        class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
        role="button"
        tabindex="0"
      >
        Avatar (Upload File)
        <input
          hidden=""
          id="avatar"
          type="file"
        />
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </label>
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="name"
          id="name-label"
        >
          name
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id="name"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-1g8cwq9"
            >
              <span>
                name
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="username"
          id="username-label"
        >
          username
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id="username"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-1g8cwq9"
            >
              <span>
                username
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="password"
          id="password-label"
        >
          password
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id="password"
            type="password"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-1g8cwq9"
            >
              <span>
                password
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="sc-gsDKAQ hmwpO"
      >
        <a
          href="/login"
        >
          already registred?
        </a>
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root Mui-disabled  css-sghohy-MuiButtonBase-root-MuiButton-root"
          disabled=""
          tabindex="-1"
          type="submit"
        >
          register
        </button>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Given a UserForm component When it's rendered with islogin true Then it should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="sc-bdvvtL cRXLZB"
      >
        <div
          class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="username"
            id="username-label"
          >
            username
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id="username"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-1g8cwq9"
              >
                <span>
                  username
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="password"
            id="password-label"
          >
            password
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
              id="password"
              type="password"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-1g8cwq9"
              >
                <span>
                  password
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="sc-gsDKAQ hmwpO"
        >
          <a
            href="/register"
          >
            not registred?
          </a>
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root Mui-disabled  css-sghohy-MuiButtonBase-root-MuiButton-root"
            disabled=""
            tabindex="-1"
            type="submit"
          >
            log in
          </button>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="sc-bdvvtL cRXLZB"
    >
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="username"
          id="username-label"
        >
          username
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id="username"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-1g8cwq9"
            >
              <span>
                username
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="password"
          id="password-label"
        >
          password
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id="password"
            type="password"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-1g8cwq9"
            >
              <span>
                password
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="sc-gsDKAQ hmwpO"
      >
        <a
          href="/register"
        >
          not registred?
        </a>
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root Mui-disabled  css-sghohy-MuiButtonBase-root-MuiButton-root"
          disabled=""
          tabindex="-1"
          type="submit"
        >
          log in
        </button>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
